File: Chapter03/Swift/LoggedInUserList.swift
//
//  LoggedInUserList.swift
//  EverydayDataStructures-Swift
//
//  Created by William Smith on 8/22/16.
//  Copyright © 2016 Websmiths, LLC. All rights reserved.
//

import Foundation

open class LoggedInUserList
{
    var _users: Array = [User]()

    init() { }
    
    func canAddUser(_ user: User) -> Bool
    {
        if (_users.contains(user) ||
            _users.count >= 30)
        {
            return false
        }
        else
        {
            return true
        }
    }
    
    open func userAuthenticated(_ user: User)
    {
        if (self.canAddUser(user))
        {
            _users.append(user)
        }
    }
    
    open func userLoggedOut(_ user: User)
    {
        if let index = _users.index(of: user)
        {
            _users.remove(at: index)
        }
    }
}

File: Chapter03/Java/LoggedInUserList.java
package main.java;

import java.util.List;
import java.util.LinkedList;


/**
 * Created by William Smith on 8/21/16.
 */
public class LoggedInUserList {

    List<User> _users;

    public LoggedInUserList()
    {
        _users = new LinkedList<User>();
    }

    boolean CanAddUser(User user)
    {
        if (_users.contains(user) ||
            _users.size() >= 30)
        {
            return false;
        }
        else
        {
            return true;
        }
    }

    public void UserAuthenticated(User user)
    {
        if (this.CanAddUser(user))
        {
            _users.add(user);
        }
    }

    public void UserLoggedOut(User user)
        {
            _users.remove(user);
        }
}
